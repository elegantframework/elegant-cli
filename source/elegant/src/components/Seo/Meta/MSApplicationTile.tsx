interface Props {
    /**
     * The directory path of the favicons.
     */
    directory?: string;
    /**
     * Display an icon for microsoft devices?
     */
    microsoftIcon?: boolean;
    /**
     * The color of the Microsoft tile.
     */
    color?: string;
};

/**
 * The head meta component for rendering microsoft application data
 * @returns A microsoft tile icon list for the seo component.
 */
const MSApplicationTile = ({
    directory = "/favicons",
    microsoftIcon = true,
    color = "#ffffff"
}: Props) => {

    let data = [];

    if(microsoftIcon){
        data.push(
            {
                property: 'msapplication-TileColor',
                content: color
            },
            {
                property: 'msapplication-TileImage',
                content: `${directory}/ms-icon-144x144.png`
            }
        );    
    };

    return data;
};

export default MSApplicationTile;