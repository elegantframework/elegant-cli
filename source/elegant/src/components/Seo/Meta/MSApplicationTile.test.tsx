import MSApplicationTile from "./MSApplicationTile";

describe('Microsoft Application Tile Component', () => {
    it('renders with data properly', () => {
        const result = MSApplicationTile({
            directory: "",
            color: "#000000"
        }); 

        expect(result).toEqual(
            expect.arrayContaining([
                expect.objectContaining({
                    content: '#000000',
                    property: 'msapplication-TileColor',
                })
            ])
        );

        expect(result).toEqual(
            expect.arrayContaining([
                expect.objectContaining({
                    content: '/ms-icon-144x144.png',
                    property: 'msapplication-TileImage',
                })
            ])
        );

        expect(result).toMatchSnapshot();
    });
});