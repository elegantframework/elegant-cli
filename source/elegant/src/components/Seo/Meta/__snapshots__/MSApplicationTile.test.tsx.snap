// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Microsoft Application Tile Component renders with data properly 1`] = `
[
  {
    "content": "#000000",
    "property": "msapplication-TileColor",
  },
  {
    "content": "/ms-icon-144x144.png",
    "property": "msapplication-TileImage",
  },
]
`;
